### PR TITLE
Use GitHub public api to fetch usernames

### PR DESCRIPTION
**Pull request by ramlmn**
Use GitHubs public api to fetch usernames instead of scraping a whole page for it.

### CODE REVIEW

File: src/libs/show-names.js [hunk 1]
```diff
@@ -9,15 +8,11 @@ const getCachedUsers = () => {
 };
 
 const fetchName = async username => {
-	// /following/you_know is the lightest page we know
-	// location.origin is required for Firefox #490
-	const pageHTML = await fetch(`${location.origin}/${username}/following`)
-		.then(res => res.text());
+	// Use GitHub public api to get username
+	const userData = await fetch(`https://api.github.com/users/${username}`)
```
Comment (busches) [inline src/libs/show-names.js]: This needs to be dynamic, to support GHE.

Comment (bfred-it) [inline src/libs/show-names.js]: But the APIs may not be available on GHE, or we might not know their hostname

**Comment by sindresorhus** [comment]
We intentionally didn't do this as the API is rate limited.

File: src/libs/show-names.js (hunk 1 above)
Comment (busches) [inline src/libs/show-names.js]: Good call, confirming that `api.<GHE-URL>` and `<GHE-URL>/api` don't work here.

**Comment by ramlmn** [comment]
Err! My bad. GitHub has a mere 60 public requests per hour limit. Closing this.

**Comment by bfred-it** [comment]
I wouldn't mind getting a token and pasting it the extension's options. OctoLinker and others do it. 

GHE seems to support it under `/api`: https://developer.github.com/v3/enterprise/

**Comment by ramlmn** [comment]
@bfred-it That would work. With a personal access token, limit extends to 5000 per hour.
The way I think about it is:
* If a token is available through settings then use it
  * If location points to GitHub use `api.github.com` endpoint else
  * use the location to get the GHE API endpoint
* If token is not available then use the scraping method.

Seems fair to me.
@sindresorhus What do you think?

**Comment by jgierer12** [comment]
If we use an API I token anyways, I suggest using the new Graph API instead

**Comment by sindresorhus** [comment]
@bfred-it Why complicate and change something that already works fine? 5000 can easily be used up too if you browse many long discussions.

**Comment by hkdobrev** [comment]
If we want to go all nuts with that we can just fetch ~10 usernames and only when you scroll to get more :)

**Comment by bfred-it** [comment]
> Why complicate and change something that already works fine?

We don't necessarily need to change old things, but new features could take advantage of faster speeds. But if GHE ends up being a problem it may not be worth it. Also definitely against having _both_ methods in as described in https://github.com/sindresorhus/refined-github/pull/581#issuecomment-313436331

> 5000 can easily be used up too if you browse many long discussions.

Caching avoids that

**Comment by ramlmn** [comment]
I would be happy if @bfred-it s webext-option-sync had support for text fields.

**Comment by bfred-it** [comment]
It does

**Comment by sindresorhus** [comment]
> but new features could take advantage of faster speeds

But the speed is not currently an issue. Nobody has complained about it.

---

I just don't see the downsides being worth the slight, but pretty unnoticeable upsides.

**Comment by ramlmn** [comment]
Is any one having issues with loading real user names on a **fresh** install of plugin?

image

I see issues with the cached names (which for a fresh install the cached names object is `undefined`).

**Comment by bfred-it** [comment]
@ramlmn unrelated.

**Comment by ramlmn** [comment]
> off topic

@bfred-it Sorry, didn't know where to ask.

**Comment by ramlmn** [comment]
Closing and moving discussion to #585.

**Pull request closed without merging**